{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "S6KIPMAF3WGR"
      },
      "source": [
        "# **Importing OMIM database and checking it with the OMIM website data**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 143,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XfeybbLYNYlY",
        "outputId": "ec7a7114-5ddc-460b-c523-4c6c901bb111"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "Zlkvroubi5kf"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Read the text file into a DataFrame\n",
        "df = pd.read_csv('/content/drive/MyDrive/pardisgene/datasets/OMIM_genemap2_chr17.txt', sep='\\t', comment='#')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 146,
      "metadata": {
        "id": "HfjIqScfjyFF"
      },
      "outputs": [],
      "source": [
        "# Function to determine inheritance based on the comment column\n",
        "def determine_inheritance(comment: str) -> str:\n",
        "    if isinstance(comment, str):  # Check if the comment is a string\n",
        "        if 'autosomal dominant' in comment.lower():\n",
        "            if 'autosomal recessive' in comment.lower():\n",
        "                return 'AD/AR'\n",
        "            else:\n",
        "                return 'AD'\n",
        "        elif 'autosomal recessive' in comment.lower():\n",
        "            return 'AR'\n",
        "        else:\n",
        "            return 'Unknown'\n",
        "    else:\n",
        "        return 'Unknown'  # Return NaN for rows with NaN values in the comment column\n",
        "\n",
        "# Add a new column for inheritance\n",
        "df['MOI'] = df['Phenotypes'].apply(determine_inheritance)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 147,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 311
        },
        "id": "YgG8x5IykKxv",
        "outputId": "d3363d42-445f-4ba8-c6ca-abed5aada46a"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 1067,\n  \"fields\": [\n    {\n      \"column\": \"Chromosome\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"chr17\"\n        ],\n        \"num_unique_values\": 1,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genomic Position Start\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 25050618,\n        \"min\": 1,\n        \"max\": 83079609,\n        \"samples\": [\n          41866917\n        ],\n        \"num_unique_values\": 1019,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genomic Position End\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 25141352,\n        \"min\": 181650,\n        \"max\": 83257441,\n        \"samples\": [\n          59215230\n        ],\n        \"num_unique_values\": 1016,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Cyto Location\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"17q24.3\"\n        ],\n        \"num_unique_values\": 101,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Computed Cyto Location\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"17q25.1\"\n        ],\n        \"num_unique_values\": 38,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MIM Number\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 162956,\n        \"min\": 100660,\n        \"max\": 620650,\n        \"samples\": [\n          610465\n        ],\n        \"num_unique_values\": 1067,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene/Locus And Other Related Symbols\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"ACSF2, FLJ20920\"\n        ],\n        \"num_unique_values\": 1067,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene Name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"Micro RNA 22\"\n        ],\n        \"num_unique_values\": 1066,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Approved Gene Symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"DNAH17\"\n        ],\n        \"num_unique_values\": 1006,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Entrez Gene ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 17776538.963555794,\n        \"min\": 15.0,\n        \"max\": 117204002.0,\n        \"samples\": [\n          1366.0\n        ],\n        \"num_unique_values\": 1041,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Ensembl Gene ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"ENSG00000006744\"\n        ],\n        \"num_unique_values\": 989,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Comments\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"mutation identified in 1 MGORS5 patient\"\n        ],\n        \"num_unique_values\": 151,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Phenotypes\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Osteogenesis imperfecta, type II, 166210 (3), Autosomal dominant; Caffey disease, 114000 (3), Autosomal dominant; Ehlers-Danlos syndrome, arthrochalasia type, 1, 130060 (3), Autosomal dominant; Osteogenesis imperfecta, type I, 166200 (3), Autosomal dominant; {Bone mineral density variation QTL, osteoporosis}, 166710 (3), Autosomal dominant; Combined osteogenesis imperfecta and Ehlers-Danlos syndrome 1, 619115 (3), Autosomal dominant; Osteogenesis imperfecta, type IV, 166220 (3), Autosomal dominant; Osteogenesis imperfecta, type III, 259420 (3), Autosomal dominant\"\n        ],\n        \"num_unique_values\": 352,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Mouse Gene Symbol/ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"Ube2o (MGI:2444266)\"\n        ],\n        \"num_unique_values\": 932,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MOI\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"AD\"\n        ],\n        \"num_unique_values\": 4,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "df"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-0ef8f8b7-c422-4a2b-9729-5fbe182b94bf\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Genomic Position Start</th>\n",
              "      <th>Genomic Position End</th>\n",
              "      <th>Cyto Location</th>\n",
              "      <th>Computed Cyto Location</th>\n",
              "      <th>MIM Number</th>\n",
              "      <th>Gene/Locus And Other Related Symbols</th>\n",
              "      <th>Gene Name</th>\n",
              "      <th>Approved Gene Symbol</th>\n",
              "      <th>Entrez Gene ID</th>\n",
              "      <th>Ensembl Gene ID</th>\n",
              "      <th>Comments</th>\n",
              "      <th>Phenotypes</th>\n",
              "      <th>Mouse Gene Symbol/ID</th>\n",
              "      <th>MOI</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>10800000</td>\n",
              "      <td>17p13</td>\n",
              "      <td>NaN</td>\n",
              "      <td>608631</td>\n",
              "      <td>ASPG2</td>\n",
              "      <td>Asperger syndrome, susceptibility to, 2</td>\n",
              "      <td>NaN</td>\n",
              "      <td>431711.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>breakpoints between CHRNE and GP1BA</td>\n",
              "      <td>{Asperger syndrome susceptibility 2}, 608631 (...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>113721</td>\n",
              "      <td>BCPR</td>\n",
              "      <td>Breast cancer-related regulator of TP53</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Breast cancer (1)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>10800000</td>\n",
              "      <td>17p13</td>\n",
              "      <td>NaN</td>\n",
              "      <td>601202</td>\n",
              "      <td>CTRCT24, CTAA2</td>\n",
              "      <td>Cataract 24, anterior polar</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1484.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Cataract 24, anterior polar, 601202 (2), Autos...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>615674</td>\n",
              "      <td>DDD3</td>\n",
              "      <td>Dowling-Degos disease 3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>102997065.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>max lod at D17S1529</td>\n",
              "      <td>Dowling-Degos disease 3, 615674 (2), Autosomal...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>613215</td>\n",
              "      <td>DUP17p13.3, C17DUPp13.3</td>\n",
              "      <td>Chromosome 17p13.3 duplication syndrome</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>includes LIS1 and/or YWHAE</td>\n",
              "      <td>Chromosome 17p13.3 duplication syndrome, 61321...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-0ef8f8b7-c422-4a2b-9729-5fbe182b94bf')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-0ef8f8b7-c422-4a2b-9729-5fbe182b94bf button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-0ef8f8b7-c422-4a2b-9729-5fbe182b94bf');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-386f1b13-16c3-4585-b984-da6cc023958c\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-386f1b13-16c3-4585-b984-da6cc023958c')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-386f1b13-16c3-4585-b984-da6cc023958c button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "  Chromosome  Genomic Position Start  Genomic Position End Cyto Location  \\\n",
              "0      chr17                       1              10800000         17p13   \n",
              "1      chr17                       1               3400000       17p13.3   \n",
              "2      chr17                       1              10800000         17p13   \n",
              "3      chr17                       1               3400000       17p13.3   \n",
              "4      chr17                       1               3400000       17p13.3   \n",
              "\n",
              "  Computed Cyto Location  MIM Number Gene/Locus And Other Related Symbols  \\\n",
              "0                    NaN      608631                                ASPG2   \n",
              "1                    NaN      113721                                 BCPR   \n",
              "2                    NaN      601202                       CTRCT24, CTAA2   \n",
              "3                    NaN      615674                                 DDD3   \n",
              "4                    NaN      613215              DUP17p13.3, C17DUPp13.3   \n",
              "\n",
              "                                 Gene Name Approved Gene Symbol  \\\n",
              "0  Asperger syndrome, susceptibility to, 2                  NaN   \n",
              "1  Breast cancer-related regulator of TP53                  NaN   \n",
              "2              Cataract 24, anterior polar                  NaN   \n",
              "3                  Dowling-Degos disease 3                  NaN   \n",
              "4  Chromosome 17p13.3 duplication syndrome                  NaN   \n",
              "\n",
              "   Entrez Gene ID Ensembl Gene ID                             Comments  \\\n",
              "0        431711.0             NaN  breakpoints between CHRNE and GP1BA   \n",
              "1             NaN             NaN                                  NaN   \n",
              "2          1484.0             NaN                                  NaN   \n",
              "3     102997065.0             NaN                  max lod at D17S1529   \n",
              "4             NaN             NaN           includes LIS1 and/or YWHAE   \n",
              "\n",
              "                                          Phenotypes Mouse Gene Symbol/ID  \\\n",
              "0  {Asperger syndrome susceptibility 2}, 608631 (...                  NaN   \n",
              "1                                  Breast cancer (1)                  NaN   \n",
              "2  Cataract 24, anterior polar, 601202 (2), Autos...                  NaN   \n",
              "3  Dowling-Degos disease 3, 615674 (2), Autosomal...                  NaN   \n",
              "4  Chromosome 17p13.3 duplication syndrome, 61321...                  NaN   \n",
              "\n",
              "       MOI  \n",
              "0  Unknown  \n",
              "1  Unknown  \n",
              "2       AD  \n",
              "3       AD  \n",
              "4  Unknown  "
            ]
          },
          "execution_count": 147,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 148,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "11VurA5SkLci",
        "outputId": "13dba569-b91e-4046-abb0-4f9a1d811060"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Unknown    756\n",
              "AR         163\n",
              "AD         114\n",
              "AD/AR       34\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 148,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 149,
      "metadata": {
        "id": "k38ylA3OnvYD"
      },
      "outputs": [],
      "source": [
        "!pip install omim"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 150,
      "metadata": {
        "id": "Gjram_ncvIk0"
      },
      "outputs": [],
      "source": [
        "df['MIM Number'] = df['MIM Number'].astype(str)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 151,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qC0gjun4kROm",
        "outputId": "807a564d-431b-424d-bf50-83ae94c29e49"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "1067it [10:45,  1.65it/s]\n"
          ]
        }
      ],
      "source": [
        "import subprocess\n",
        "import re\n",
        "import omim\n",
        "from tqdm import tqdm\n",
        "\n",
        "def get_inheritance_mode(mim_number: str) -> str:\n",
        "    \"\"\"\n",
        "    Retrieves the inheritance mode for a given MIM (Mendelian Inheritance in Man) number.\n",
        "\n",
        "    Parameters:\n",
        "        mim_number : str\n",
        "            The MIM number for which the inheritance mode is to be retrieved.\n",
        "\n",
        "    Returns:\n",
        "        str\n",
        "            The inheritance mode corresponding to the provided MIM number. Possible values are \"AR\" (Autosomal Recessive),\n",
        "            \"AD\" (Autosomal Dominant), \"AD/AR\" (Autosomal Dominant and Recessive), \"AR/AD\" (Autosomal Recessive and Dominant),\n",
        "            or \"Unknown\" if the inheritance mode could not be retrieved.\n",
        "\n",
        "    This function utilizes the 'omim' command-line tool to query the OMIM database and extract the inheritance mode\n",
        "    information for a given MIM number. It executes the command and captures the output, then extracts the inheritance\n",
        "    mode using regular expressions. If the inheritance mode is not one of the predefined values (\"AR\", \"AD\", \"AD/AR\", \"AR/AD\"),\n",
        "    it is marked as \"Unknown\". If an error occurs during the retrieval process, it returns \"Error retrieving inheritance mode\".\n",
        "\n",
        "    \"\"\"\n",
        "    try:\n",
        "        # Execute the command and capture its output\n",
        "        output_bytes = subprocess.check_output([\"omim\", \"query\", \"-s\", \"mim_number\", mim_number])\n",
        "        output = output_bytes.decode(\"utf-8\")\n",
        "\n",
        "        # Extract the inheritance mode from the output\n",
        "        inheritance_mode_match = re.search(r'\"Inheritance\": \"(\\w+)\"', output)\n",
        "        if inheritance_mode_match:\n",
        "            inheritance_mode = inheritance_mode_match.group(1)\n",
        "            if inheritance_mode not in ['AR', 'AD', 'AD/AR', 'AR/AD']:\n",
        "                inheritance_mode = \"Unknown\"\n",
        "        else:\n",
        "            inheritance_mode = \"Unknown\"\n",
        "        return inheritance_mode\n",
        "    except subprocess.CalledProcessError:\n",
        "        return \"Error retrieving inheritance mode\"\n",
        "\n",
        "\n",
        "# Iterate over rows of the DataFrame\n",
        "for index, row in tqdm(df.iterrows()):\n",
        "    # Check if the \"inheritance2\" column has the value \"Unknown\"\n",
        "    if row['MOI'] == \"Unknown\":\n",
        "        # Call the function to get the inheritance mode\n",
        "        inheritance_mode = get_inheritance_mode(row['MIM Number'])\n",
        "        # Update the \"inheritance2\" column with the retrieved inheritance mode\n",
        "        df.at[index, 'MOI'] = inheritance_mode\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 152,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6sjsGHrm_3gl",
        "outputId": "410b4f3a-aacc-4080-a7d5-628d603207b1"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Unknown    754\n",
              "AR         163\n",
              "AD         116\n",
              "AD/AR       34\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 152,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 171,
      "metadata": {
        "id": "zt-Duyd98yds"
      },
      "outputs": [],
      "source": [
        "def add_new_data(df: pd.DataFrame , db_df: pd.DataFrame, gene_column_name: str, moi_column_name: str) -> None:\n",
        "  \"\"\"\n",
        "  Updates the Mode of Inheritance (MOI) column in a DataFrame (df) based on information from another DataFrame (db_df).\n",
        "\n",
        "  Parameters:\n",
        "      df : pd.DataFrame\n",
        "          The DataFrame containing the data to be updated.\n",
        "\n",
        "      db_df : pd.DataFrame\n",
        "          The DataFrame containing the reference data used for updating.\n",
        "\n",
        "      gene_column_name : str\n",
        "          The name of the column in db_df containing gene names.\n",
        "\n",
        "      moi_column_name : str\n",
        "          The name of the column in db_df containing Mode of Inheritance values.\n",
        "\n",
        "  Returns:\n",
        "      None\n",
        "\n",
        "  This function iterates over each row in the input DataFrame (df). For each row where the Mode of Inheritance (MOI)\n",
        "  value is 'Unknown', it extracts gene names from the 'Gene/Locus And Other Related Symbols' column and adds the 'Approved Gene Symbol'\n",
        "  to the list if it's available. Then, it searches for each gene name in the reference DataFrame (db_df) using the specified\n",
        "  gene column name. If a match is found, it updates the MOI value in the input DataFrame (df) with the corresponding value from\n",
        "  the reference DataFrame (db_df). The search is performed based on the provided gene_column_name and moi_column_name.\n",
        "\n",
        "  \"\"\"\n",
        "\n",
        "  for index, row in tqdm(df.iterrows()):\n",
        "    # Check if the Inheritance value is 'Unknown'\n",
        "    if row['MOI'] == 'Unknown':\n",
        "        # Split the gene names if there are multiple names\n",
        "        gene_names = row['Gene/Locus And Other Related Symbols'].split(',')\n",
        "\n",
        "        # Add HGNC to gene_names list\n",
        "        if type(row['Approved Gene Symbol']) == str and row['Approved Gene Symbol'] not in gene_names:\n",
        "            gene_names.append(row['Approved Gene Symbol'])\n",
        "\n",
        "        # Iterate over each gene name\n",
        "        for gene_name in gene_names:\n",
        "            # Search for the gene in db_df\n",
        "            matching_row = db_df[db_df[gene_column_name] == gene_name.strip()]  # Strip to remove any leading/trailing spaces\n",
        "            if not matching_row.empty:\n",
        "                # Update Inheritance column in df with the value from db_df\n",
        "                if matching_row[moi_column_name].values[0].upper() in ['AR', 'AD', 'AD/AR', 'AR/AD']:\n",
        "                    df.at[index, 'MOI'] = matching_row[moi_column_name].values[0].upper()\n",
        "                break  # Break out of the loop if a match is found for any gene name"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0cbVKwiQ3EUC"
      },
      "source": [
        "# **Importing CGD database and concating with the OMIM database**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 155,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 311
        },
        "id": "7iNUyh-PvKd6",
        "outputId": "096f38bb-d549-462b-e5f0-a76751e4806b"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"cgd_db\",\n  \"rows\": 4680,\n  \"fields\": [\n    {\n      \"column\": \"GENE\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"SLC2A10\",\n          \"ABHD16A\",\n          \"ITGB6\"\n        ],\n        \"num_unique_values\": 4680,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"HGNC ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9118,\n        \"min\": 1,\n        \"max\": 55085,\n        \"samples\": [\n          13444,\n          13921,\n          6161\n        ],\n        \"num_unique_values\": 4680,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ENTREZ GENE ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 29839254,\n        \"min\": 2,\n        \"max\": 2000000000,\n        \"samples\": [\n          81031,\n          7920,\n          3694\n        ],\n        \"num_unique_values\": 4680,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"CONDITION\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"Noonan syndrome 1; LEOPARD syndrome 1\",\n          \"CYP2A6-related drug metabolism\",\n          \"Coronary artery disease, autosomal dominant 2\"\n        ],\n        \"num_unique_values\": 4628,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"INHERITANCE\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Digenic\",\n          \"AD (with imprinting)\",\n          \"AD\"\n        ],\n        \"num_unique_values\": 25,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"AGE GROUP\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Pediatric (Lymphatic malformation 13); Adult (Aneurysm, intracranial berry, 12)\",\n          \"Adult (Colorectal cancer, hereditary nonpolyposis, type 1; Endometrial cancer; Muir-Torre syndrome)/Pediatric (Mismatch repair cancer syndrome)\",\n          \"Adult (Paragangliomas 5; Gastrointestinal stromal tumors); Pediatric (Cardiomyopathy, dilated, 1GG; Neurodegeneration with ataxia and late-onset optic atrophy; Leigh syndrome/Mitochondrial respiratory chain complex II deficiency)\"\n        ],\n        \"num_unique_values\": 31,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALLELIC CONDITIONS\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Allelic with Brain small vessel disease 2 (AD)\",\n          \"Allelic with Krabbe disease, atypical (AR); Combined saposin deficiency (AR); Gaucher disease, atypical, due to saposin C deficiency (AR); Metachromatic leukodystrophy due to saposin-b deficiency (AR)\",\n          \"Allelic with Amyotrophic lateral sclerosis 12 with or without frontotemporal dementia (AD/AR)\"\n        ],\n        \"num_unique_values\": 396,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MANIFESTATION CATEGORIES\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Cardiovascular; Craniofacial; Endocrine; Musculoskeletal; Neurologic\",\n          \"Biochemical; Hematologic; Neurologic\",\n          \"Craniofacial; Dental; Endocrine; Musculoskeletal; Neurologic\"\n        ],\n        \"num_unique_values\": 1117,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"INTERVENTION CATEGORIES\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Hematologic; Oncologic\",\n          \"Dermatologic\",\n          \"Obstetric; Oncologic\"\n        ],\n        \"num_unique_values\": 273,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"COMMENTS\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Elevated histidine-rich glycoprotein (HRG) has also been described as associated with risk of thromboembolism\",\n          \"Renal transplantation has been described\",\n          \"Unlike hypobetalipoproteinemia due to variants in APOB, individuals with variants in ANGPTL3 do not appear to to have increased susceptibility to hepatic steatosis or gastrointestinal or cardiovascular manifestations compared to unaffected relatives; Heterozygous individuals may demonstrate laboratory-based manifestations (eg, lower plasma LDL and triglycerides)\"\n        ],\n        \"num_unique_values\": 637,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"INTERVENTION/RATIONALE\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Levodopa can be beneficial in individuals with Parkinson disease\",\n          \"The condition may not be clinically relevant\",\n          \"Among other features, early recognition and treatment of hearing impairment may improve outcomes, including speech and language development\"\n        ],\n        \"num_unique_values\": 1903,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REFERENCES\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          22541560,\n          \"22279524; 23035047; 25319849; 25500575; 26483087; 26494257; 27282546; 37344571\",\n          22693284\n        ],\n        \"num_unique_values\": 4621,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "cgd_db"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-d739e125-38ce-4d9e-8499-bee3aef6a2eb\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GENE</th>\n",
              "      <th>HGNC ID</th>\n",
              "      <th>ENTREZ GENE ID</th>\n",
              "      <th>CONDITION</th>\n",
              "      <th>INHERITANCE</th>\n",
              "      <th>AGE GROUP</th>\n",
              "      <th>ALLELIC CONDITIONS</th>\n",
              "      <th>MANIFESTATION CATEGORIES</th>\n",
              "      <th>INTERVENTION CATEGORIES</th>\n",
              "      <th>COMMENTS</th>\n",
              "      <th>INTERVENTION/RATIONALE</th>\n",
              "      <th>REFERENCES</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>A2M</td>\n",
              "      <td>7</td>\n",
              "      <td>2</td>\n",
              "      <td>Alpha-2-macroglobulin deficiency</td>\n",
              "      <td>AD</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>General</td>\n",
              "      <td>General</td>\n",
              "      <td>Variants have been implicated in pulmonary dis...</td>\n",
              "      <td>The clinical consequences of variants are unclear</td>\n",
              "      <td>94459; 2475424; 1370808</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>A2ML1</td>\n",
              "      <td>23336</td>\n",
              "      <td>144568</td>\n",
              "      <td>Otitis media, susceptibility to</td>\n",
              "      <td>AD</td>\n",
              "      <td>Pediatric</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Allergy/Immunology/Infectious</td>\n",
              "      <td>Allergy/Immunology/Infectious</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Individuals may have increased susceptibility ...</td>\n",
              "      <td>26121085</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>A4GALT</td>\n",
              "      <td>18149</td>\n",
              "      <td>53947</td>\n",
              "      <td>Blood group, P1PK system</td>\n",
              "      <td>BG</td>\n",
              "      <td>Pediatric</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Hematologic</td>\n",
              "      <td>Hematologic</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Variants associated with a blood group may be ...</td>\n",
              "      <td>10993874</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>AAAS</td>\n",
              "      <td>13666</td>\n",
              "      <td>8086</td>\n",
              "      <td>Achalasia-addisonianism-alacrimia syndrome</td>\n",
              "      <td>AR</td>\n",
              "      <td>Pediatric</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Dermatologic; Endocrine; Gastrointestinal; Neu...</td>\n",
              "      <td>Endocrine</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Surveillance and treatment/preventive measures...</td>\n",
              "      <td>78049; 6243664; 3565479; 1537368; 8006362; 789...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>AAGAB</td>\n",
              "      <td>25662</td>\n",
              "      <td>79719</td>\n",
              "      <td>Keratoderma, palmoplantar, punctate type IA</td>\n",
              "      <td>AD</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Dermatologic</td>\n",
              "      <td>General</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Genetic knowledge may be beneficial related to...</td>\n",
              "      <td>23000146; 23064416</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d739e125-38ce-4d9e-8499-bee3aef6a2eb')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-d739e125-38ce-4d9e-8499-bee3aef6a2eb button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-d739e125-38ce-4d9e-8499-bee3aef6a2eb');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-ba53fcd8-de36-47b5-ae3c-1f947b5abc39\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ba53fcd8-de36-47b5-ae3c-1f947b5abc39')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-ba53fcd8-de36-47b5-ae3c-1f947b5abc39 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "     GENE  HGNC ID  ENTREZ GENE ID  \\\n",
              "0     A2M        7               2   \n",
              "1   A2ML1    23336          144568   \n",
              "2  A4GALT    18149           53947   \n",
              "3    AAAS    13666            8086   \n",
              "4   AAGAB    25662           79719   \n",
              "\n",
              "                                     CONDITION INHERITANCE  AGE GROUP  \\\n",
              "0             Alpha-2-macroglobulin deficiency          AD        NaN   \n",
              "1              Otitis media, susceptibility to          AD  Pediatric   \n",
              "2                     Blood group, P1PK system          BG  Pediatric   \n",
              "3   Achalasia-addisonianism-alacrimia syndrome          AR  Pediatric   \n",
              "4  Keratoderma, palmoplantar, punctate type IA          AD        NaN   \n",
              "\n",
              "  ALLELIC CONDITIONS                           MANIFESTATION CATEGORIES  \\\n",
              "0                NaN                                            General   \n",
              "1                NaN                      Allergy/Immunology/Infectious   \n",
              "2                NaN                                        Hematologic   \n",
              "3                NaN  Dermatologic; Endocrine; Gastrointestinal; Neu...   \n",
              "4                NaN                                       Dermatologic   \n",
              "\n",
              "         INTERVENTION CATEGORIES  \\\n",
              "0                        General   \n",
              "1  Allergy/Immunology/Infectious   \n",
              "2                    Hematologic   \n",
              "3                      Endocrine   \n",
              "4                        General   \n",
              "\n",
              "                                            COMMENTS  \\\n",
              "0  Variants have been implicated in pulmonary dis...   \n",
              "1                                                NaN   \n",
              "2                                                NaN   \n",
              "3                                                NaN   \n",
              "4                                                NaN   \n",
              "\n",
              "                              INTERVENTION/RATIONALE  \\\n",
              "0  The clinical consequences of variants are unclear   \n",
              "1  Individuals may have increased susceptibility ...   \n",
              "2  Variants associated with a blood group may be ...   \n",
              "3  Surveillance and treatment/preventive measures...   \n",
              "4  Genetic knowledge may be beneficial related to...   \n",
              "\n",
              "                                          REFERENCES  \n",
              "0                            94459; 2475424; 1370808  \n",
              "1                                           26121085  \n",
              "2                                           10993874  \n",
              "3  78049; 6243664; 3565479; 1537368; 8006362; 789...  \n",
              "4                                 23000146; 23064416  "
            ]
          },
          "execution_count": 155,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "cgd_db = pd.read_excel('/content/drive/MyDrive/pardisgene/datasets/CGD all.xls')\n",
        "cgd_db.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 156,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-fuwN-gD16pE",
        "outputId": "b7e39ed1-335d-4687-d3e1-bfeaa73c4e0e"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "AR                                                                                                    2470\n",
              "AD                                                                                                    1378\n",
              "AD/AR                                                                                                  439\n",
              "XL                                                                                                     249\n",
              "BG                                                                                                      31\n",
              "Maternal                                                                                                28\n",
              "AD/AR/Digenic                                                                                           21\n",
              "AR/Digenic                                                                                              17\n",
              "AD/Digenic                                                                                              13\n",
              "Digenic                                                                                                  8\n",
              "BG/AR                                                                                                    4\n",
              "AD/AR/Digenic (with CFTR or other SCCN1 genes)                                                           3\n",
              "YL                                                                                                       3\n",
              "AD/BG                                                                                                    3\n",
              "XL (involving both OPN1 genes)                                                                           2\n",
              "AD/AR/BG                                                                                                 2\n",
              "AD/Multigenic                                                                                            1\n",
              "AD/Digenic/Multigenic                                                                                    1\n",
              "AD (with imprinting)                                                                                     1\n",
              "AD/AR/Digenic/Multigenic                                                                                 1\n",
              "AD/AR (Triallelic)                                                                                       1\n",
              "AD/Digenic (Severe digenic insulin resistance can be due to digenic variants in PPP1R3A and PPARG)       1\n",
              "XL/Digenic                                                                                               1\n",
              "XL/PAR                                                                                                   1\n",
              "Ad                                                                                                       1\n",
              "Name: INHERITANCE, dtype: int64"
            ]
          },
          "execution_count": 156,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "cgd_db['INHERITANCE'].value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 157,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BvUsuboY7fId",
        "outputId": "52acafe4-f034-4572-d528-ea60c5a7105a"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "1067it [00:01, 784.48it/s]\n"
          ]
        }
      ],
      "source": [
        "add_new_data(df, cgd_db, 'GENE', 'INHERITANCE')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 158,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qEhvjQ7pAxbb",
        "outputId": "d96ce35b-4366-4618-96dc-abea66e350be"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Unknown    738\n",
              "AR         168\n",
              "AD         126\n",
              "AD/AR       35\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 158,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ktf4_B-W4SQD"
      },
      "source": [
        "# **Importing Clingen database and concating with the OMIM, CGD database**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 159,
      "metadata": {
        "id": "nB_3lAzl4Px5"
      },
      "outputs": [],
      "source": [
        "clingen_db = pd.read_csv(\"/content/drive/MyDrive/pardisgene/datasets/Clingen-Gene-Disease-Summary-2024-02-12.csv\", sep=',', comment='#')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 160,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "Zr_t7YTQ4Psj",
        "outputId": "c2ccec49-cd15-4a2e-c29e-50853e967e87"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"clingen_db\",\n  \"rows\": 2489,\n  \"fields\": [\n    {\n      \"column\": \"GENE SYMBOL\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"CDH3\",\n          \"NOD2\",\n          \"CNGA1\"\n        ],\n        \"num_unique_values\": 2044,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENE ID (HGNC)\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"HGNC:1762\",\n          \"HGNC:5331\",\n          \"HGNC:2148\"\n        ],\n        \"num_unique_values\": 2044,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"DISEASE LABEL\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"ciliopathy\",\n          \"muscle-eye-brain disease\",\n          \"hypotonia, infantile, with psychomotor retardation and characteristic facies 2\"\n        ],\n        \"num_unique_values\": 1360,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"DISEASE ID (MONDO)\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"MONDO:0005308\",\n          \"MONDO:0018939\",\n          \"MONDO:0014777\"\n        ],\n        \"num_unique_values\": 1360,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MOI\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"AD\",\n          \"AR\",\n          \"MT\"\n        ],\n        \"num_unique_values\": 6,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"SOP\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"SOP5\",\n          \"SOP8\",\n          \"SOP6\"\n        ],\n        \"num_unique_values\": 7,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"CLASSIFICATION\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Disputed\",\n          \"Definitive\",\n          \"Strong\"\n        ],\n        \"num_unique_values\": 7,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ONLINE REPORT\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"https://search.clinicalgenome.org/kb/gene-validity/CGGV:assertion_60c28393-bd46-4da4-9700-ce2fb005377b-2023-07-14T160000.000Z\",\n          \"https://search.clinicalgenome.org/kb/gene-validity/CGGV:assertion_a51b8f9f-0e11-4d82-b454-15d27e974dd7-2021-06-24T204605.941Z\",\n          \"https://search.clinicalgenome.org/kb/gene-validity/CGGV:assertion_2d6f7f67-63e7-4587-9216-6909f5883de9-2023-02-27T190000.000Z\"\n        ],\n        \"num_unique_values\": 2489,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"CLASSIFICATION DATE\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"min\": \"2016-02-15T00:00:00\",\n        \"max\": \"2024-02-07T17:00:00.000Z\",\n        \"samples\": [\n          \"2023-11-02T16:00:00.000Z\",\n          \"2020-12-18T20:05:20.841Z\",\n          \"2020-06-01T16:00:00.000Z\"\n        ],\n        \"num_unique_values\": 1573,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GCEP\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"RASopathy\",\n          \"General Gene Curation\",\n          \"Amyotrophic Lateral Sclerosis Spectrum Disorders\"\n        ],\n        \"num_unique_values\": 56,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "clingen_db"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-e63ab044-7b4d-4e73-ba7a-973fc85e4ef6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GENE SYMBOL</th>\n",
              "      <th>GENE ID (HGNC)</th>\n",
              "      <th>DISEASE LABEL</th>\n",
              "      <th>DISEASE ID (MONDO)</th>\n",
              "      <th>MOI</th>\n",
              "      <th>SOP</th>\n",
              "      <th>CLASSIFICATION</th>\n",
              "      <th>ONLINE REPORT</th>\n",
              "      <th>CLASSIFICATION DATE</th>\n",
              "      <th>GCEP</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>A2ML1</td>\n",
              "      <td>HGNC:23336</td>\n",
              "      <td>Noonan syndrome</td>\n",
              "      <td>MONDO:0018997</td>\n",
              "      <td>AD</td>\n",
              "      <td>SOP5</td>\n",
              "      <td>Disputed</td>\n",
              "      <td>https://search.clinicalgenome.org/kb/gene-vali...</td>\n",
              "      <td>2018-06-07T16:00:00.000Z</td>\n",
              "      <td>RASopathy</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>AARS2</td>\n",
              "      <td>HGNC:21022</td>\n",
              "      <td>mitochondrial disease</td>\n",
              "      <td>MONDO:0044970</td>\n",
              "      <td>AR</td>\n",
              "      <td>SOP8</td>\n",
              "      <td>Definitive</td>\n",
              "      <td>https://search.clinicalgenome.org/kb/gene-vali...</td>\n",
              "      <td>2022-04-18T16:00:00.000Z</td>\n",
              "      <td>Mitochondrial Diseases</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>AASS</td>\n",
              "      <td>HGNC:17366</td>\n",
              "      <td>hyperlysinemia</td>\n",
              "      <td>MONDO:0009388</td>\n",
              "      <td>AR</td>\n",
              "      <td>SOP9</td>\n",
              "      <td>Definitive</td>\n",
              "      <td>https://search.clinicalgenome.org/kb/gene-vali...</td>\n",
              "      <td>2022-10-14T16:00:00.000Z</td>\n",
              "      <td>Aminoacidopathy</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ABAT</td>\n",
              "      <td>HGNC:23</td>\n",
              "      <td>developmental and epileptic encephalopathy</td>\n",
              "      <td>MONDO:0100062</td>\n",
              "      <td>AR</td>\n",
              "      <td>SOP8</td>\n",
              "      <td>Moderate</td>\n",
              "      <td>https://search.clinicalgenome.org/kb/gene-vali...</td>\n",
              "      <td>2022-04-19T16:00:00.000Z</td>\n",
              "      <td>Epilepsy</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ABCA4</td>\n",
              "      <td>HGNC:34</td>\n",
              "      <td>ABCA4-related retinopathy</td>\n",
              "      <td>MONDO:0800406</td>\n",
              "      <td>AR</td>\n",
              "      <td>SOP9</td>\n",
              "      <td>Definitive</td>\n",
              "      <td>https://search.clinicalgenome.org/kb/gene-vali...</td>\n",
              "      <td>2022-10-06T16:00:00.000Z</td>\n",
              "      <td>Retina</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e63ab044-7b4d-4e73-ba7a-973fc85e4ef6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e63ab044-7b4d-4e73-ba7a-973fc85e4ef6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e63ab044-7b4d-4e73-ba7a-973fc85e4ef6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-8e6a9fe6-6ef2-49c3-8125-b580239ffea8\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-8e6a9fe6-6ef2-49c3-8125-b580239ffea8')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-8e6a9fe6-6ef2-49c3-8125-b580239ffea8 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "  GENE SYMBOL GENE ID (HGNC)                               DISEASE LABEL  \\\n",
              "0       A2ML1     HGNC:23336                             Noonan syndrome   \n",
              "1       AARS2     HGNC:21022                       mitochondrial disease   \n",
              "2        AASS     HGNC:17366                              hyperlysinemia   \n",
              "3        ABAT        HGNC:23  developmental and epileptic encephalopathy   \n",
              "4       ABCA4        HGNC:34                   ABCA4-related retinopathy   \n",
              "\n",
              "  DISEASE ID (MONDO) MOI   SOP CLASSIFICATION  \\\n",
              "0      MONDO:0018997  AD  SOP5       Disputed   \n",
              "1      MONDO:0044970  AR  SOP8     Definitive   \n",
              "2      MONDO:0009388  AR  SOP9     Definitive   \n",
              "3      MONDO:0100062  AR  SOP8       Moderate   \n",
              "4      MONDO:0800406  AR  SOP9     Definitive   \n",
              "\n",
              "                                       ONLINE REPORT  \\\n",
              "0  https://search.clinicalgenome.org/kb/gene-vali...   \n",
              "1  https://search.clinicalgenome.org/kb/gene-vali...   \n",
              "2  https://search.clinicalgenome.org/kb/gene-vali...   \n",
              "3  https://search.clinicalgenome.org/kb/gene-vali...   \n",
              "4  https://search.clinicalgenome.org/kb/gene-vali...   \n",
              "\n",
              "        CLASSIFICATION DATE                    GCEP  \n",
              "0  2018-06-07T16:00:00.000Z               RASopathy  \n",
              "1  2022-04-18T16:00:00.000Z  Mitochondrial Diseases  \n",
              "2  2022-10-14T16:00:00.000Z         Aminoacidopathy  \n",
              "3  2022-04-19T16:00:00.000Z                Epilepsy  \n",
              "4  2022-10-06T16:00:00.000Z                  Retina  "
            ]
          },
          "execution_count": 160,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "clingen_db.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 161,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hJ8YR7JJ4Pp_",
        "outputId": "0063d9fd-872c-44ac-c933-63e628ba8891"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "AR              1155\n",
              "AD              1067\n",
              "XL               164\n",
              "MT                47\n",
              "SD                41\n",
              "Undetermined      15\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 161,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "clingen_db['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 162,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kwreB0qjA7Td",
        "outputId": "15268d8f-e7b9-4d13-a7c4-0cf514d0b274"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "1067it [00:00, 1146.40it/s]\n"
          ]
        }
      ],
      "source": [
        "add_new_data(df, clingen_db, 'GENE SYMBOL', 'MOI')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 163,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XT975BRGBBZN",
        "outputId": "8daa5fee-9069-4c2a-d756-bbc7bc7a8875"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Unknown    731\n",
              "AR         169\n",
              "AD         132\n",
              "AD/AR       35\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 163,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "BClpbeuJ7b9Y"
      },
      "source": [
        "# **Importing GENCC database and concating with the existing database**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 164,
      "metadata": {
        "id": "XlUk789L7abj"
      },
      "outputs": [],
      "source": [
        "gencc_db = pd.read_excel(\"/content/drive/MyDrive/pardisgene/datasets/gencc-submissions.xlsx\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 165,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 516
        },
        "id": "V4cLP6_C7pNJ",
        "outputId": "87769f0f-7ab2-4384-8116-42cc89a8e6d7"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "gencc_db"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-fec104f6-93cb-4c49-9839-fd047041e791\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>uuid</th>\n",
              "      <th>gene_curie</th>\n",
              "      <th>gene_symbol</th>\n",
              "      <th>disease_curie</th>\n",
              "      <th>disease_title</th>\n",
              "      <th>disease_original_curie</th>\n",
              "      <th>disease_original_title</th>\n",
              "      <th>classification_curie</th>\n",
              "      <th>classification_title</th>\n",
              "      <th>moi_curie</th>\n",
              "      <th>...</th>\n",
              "      <th>submitted_as_submitter_name</th>\n",
              "      <th>submitted_as_classification_id</th>\n",
              "      <th>submitted_as_classification_name</th>\n",
              "      <th>submitted_as_date</th>\n",
              "      <th>submitted_as_public_report_url</th>\n",
              "      <th>submitted_as_notes</th>\n",
              "      <th>submitted_as_pmids</th>\n",
              "      <th>submitted_as_assertion_criteria_url</th>\n",
              "      <th>submitted_as_submission_id</th>\n",
              "      <th>submitted_run_date</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GENCC_000101-HGNC_10896-OMIM_182212-HP_0000006...</td>\n",
              "      <td>HGNC:10896</td>\n",
              "      <td>SKI</td>\n",
              "      <td>MONDO:0008426</td>\n",
              "      <td>Shprintzen-Goldberg syndrome</td>\n",
              "      <td>OMIM:182212</td>\n",
              "      <td>Shprintzen-Goldberg syndrome</td>\n",
              "      <td>GENCC:100001</td>\n",
              "      <td>Definitive</td>\n",
              "      <td>HP:0000006</td>\n",
              "      <td>...</td>\n",
              "      <td>Ambry Genetics</td>\n",
              "      <td>GENCC:100001</td>\n",
              "      <td>Definitive</td>\n",
              "      <td>2018-03-30 13:31:56</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PMID:28106320</td>\n",
              "      <td>1034</td>\n",
              "      <td>2020-12-24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GENCC_000101-HGNC_16636-OMIM_171300-HP_0000006...</td>\n",
              "      <td>HGNC:16636</td>\n",
              "      <td>KIF1B</td>\n",
              "      <td>MONDO:0008233</td>\n",
              "      <td>pheochromocytoma</td>\n",
              "      <td>OMIM:171300</td>\n",
              "      <td>{Pheochromocytoma, susceptibility to}</td>\n",
              "      <td>GENCC:100003</td>\n",
              "      <td>Moderate</td>\n",
              "      <td>HP:0000006</td>\n",
              "      <td>...</td>\n",
              "      <td>Ambry Genetics</td>\n",
              "      <td>GENCC:100003</td>\n",
              "      <td>Moderate</td>\n",
              "      <td>2019-12-04 13:30:43</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PMID:28106320</td>\n",
              "      <td>69237</td>\n",
              "      <td>2020-12-24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GENCC_000101-HGNC_16636-OMIM_118210-HP_0000006...</td>\n",
              "      <td>HGNC:16636</td>\n",
              "      <td>KIF1B</td>\n",
              "      <td>MONDO:0007308</td>\n",
              "      <td>Charcot-Marie-Tooth disease type 2A1</td>\n",
              "      <td>OMIM:118210</td>\n",
              "      <td>Charcot-Marie-Tooth disease, type 2A1</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>HP:0000006</td>\n",
              "      <td>...</td>\n",
              "      <td>Ambry Genetics</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>2022-09-02 00:00:00</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://onlinelibrary.wiley.com/doi/10.1002/hu...</td>\n",
              "      <td>61327</td>\n",
              "      <td>2023-08-19</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GENCC_000101-HGNC_17939-OMIM_617532-HP_0000007...</td>\n",
              "      <td>HGNC:17939</td>\n",
              "      <td>SLC45A1</td>\n",
              "      <td>MONDO:0044322</td>\n",
              "      <td>intellectual developmental disorder with neuro...</td>\n",
              "      <td>OMIM:617532</td>\n",
              "      <td>Intellectual developmental disorder with neuro...</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>HP:0000007</td>\n",
              "      <td>...</td>\n",
              "      <td>Ambry Genetics</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>2020-06-26 13:32:00</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PMID:28106320</td>\n",
              "      <td>72178</td>\n",
              "      <td>2020-12-24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GENCC_000101-HGNC_11071-OMIM_616291-HP_0000007...</td>\n",
              "      <td>HGNC:11071</td>\n",
              "      <td>SLC9A1</td>\n",
              "      <td>MONDO:0014572</td>\n",
              "      <td>Lichtenstein-Knorr syndrome</td>\n",
              "      <td>OMIM:616291</td>\n",
              "      <td>Lichtenstein-Knorr syndrome</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>HP:0000007</td>\n",
              "      <td>...</td>\n",
              "      <td>Ambry Genetics</td>\n",
              "      <td>GENCC:100004</td>\n",
              "      <td>Limited</td>\n",
              "      <td>2018-08-31 13:32:02</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PMID:28106320</td>\n",
              "      <td>1705</td>\n",
              "      <td>2020-12-24</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows  30 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-fec104f6-93cb-4c49-9839-fd047041e791')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-fec104f6-93cb-4c49-9839-fd047041e791 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-fec104f6-93cb-4c49-9839-fd047041e791');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-a871e096-52ec-4364-9c1c-4b531bf34dcd\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-a871e096-52ec-4364-9c1c-4b531bf34dcd')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-a871e096-52ec-4364-9c1c-4b531bf34dcd button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                                                uuid  gene_curie gene_symbol  \\\n",
              "0  GENCC_000101-HGNC_10896-OMIM_182212-HP_0000006...  HGNC:10896         SKI   \n",
              "1  GENCC_000101-HGNC_16636-OMIM_171300-HP_0000006...  HGNC:16636       KIF1B   \n",
              "2  GENCC_000101-HGNC_16636-OMIM_118210-HP_0000006...  HGNC:16636       KIF1B   \n",
              "3  GENCC_000101-HGNC_17939-OMIM_617532-HP_0000007...  HGNC:17939     SLC45A1   \n",
              "4  GENCC_000101-HGNC_11071-OMIM_616291-HP_0000007...  HGNC:11071      SLC9A1   \n",
              "\n",
              "   disease_curie                                      disease_title  \\\n",
              "0  MONDO:0008426                       Shprintzen-Goldberg syndrome   \n",
              "1  MONDO:0008233                                   pheochromocytoma   \n",
              "2  MONDO:0007308               Charcot-Marie-Tooth disease type 2A1   \n",
              "3  MONDO:0044322  intellectual developmental disorder with neuro...   \n",
              "4  MONDO:0014572                        Lichtenstein-Knorr syndrome   \n",
              "\n",
              "  disease_original_curie                             disease_original_title  \\\n",
              "0            OMIM:182212                       Shprintzen-Goldberg syndrome   \n",
              "1            OMIM:171300              {Pheochromocytoma, susceptibility to}   \n",
              "2            OMIM:118210              Charcot-Marie-Tooth disease, type 2A1   \n",
              "3            OMIM:617532  Intellectual developmental disorder with neuro...   \n",
              "4            OMIM:616291                        Lichtenstein-Knorr syndrome   \n",
              "\n",
              "  classification_curie classification_title   moi_curie  ...  \\\n",
              "0         GENCC:100001           Definitive  HP:0000006  ...   \n",
              "1         GENCC:100003             Moderate  HP:0000006  ...   \n",
              "2         GENCC:100004              Limited  HP:0000006  ...   \n",
              "3         GENCC:100004              Limited  HP:0000007  ...   \n",
              "4         GENCC:100004              Limited  HP:0000007  ...   \n",
              "\n",
              "  submitted_as_submitter_name submitted_as_classification_id  \\\n",
              "0              Ambry Genetics                   GENCC:100001   \n",
              "1              Ambry Genetics                   GENCC:100003   \n",
              "2              Ambry Genetics                   GENCC:100004   \n",
              "3              Ambry Genetics                   GENCC:100004   \n",
              "4              Ambry Genetics                   GENCC:100004   \n",
              "\n",
              "  submitted_as_classification_name    submitted_as_date  \\\n",
              "0                       Definitive  2018-03-30 13:31:56   \n",
              "1                         Moderate  2019-12-04 13:30:43   \n",
              "2                          Limited  2022-09-02 00:00:00   \n",
              "3                          Limited  2020-06-26 13:32:00   \n",
              "4                          Limited  2018-08-31 13:32:02   \n",
              "\n",
              "  submitted_as_public_report_url submitted_as_notes submitted_as_pmids  \\\n",
              "0                            NaN                NaN                NaN   \n",
              "1                            NaN                NaN                NaN   \n",
              "2                            NaN                NaN                NaN   \n",
              "3                            NaN                NaN                NaN   \n",
              "4                            NaN                NaN                NaN   \n",
              "\n",
              "                 submitted_as_assertion_criteria_url  \\\n",
              "0                                     PMID:28106320   \n",
              "1                                     PMID:28106320   \n",
              "2  https://onlinelibrary.wiley.com/doi/10.1002/hu...   \n",
              "3                                     PMID:28106320   \n",
              "4                                     PMID:28106320   \n",
              "\n",
              "  submitted_as_submission_id submitted_run_date  \n",
              "0                       1034         2020-12-24  \n",
              "1                      69237         2020-12-24  \n",
              "2                      61327         2023-08-19  \n",
              "3                      72178         2020-12-24  \n",
              "4                       1705         2020-12-24  \n",
              "\n",
              "[5 rows x 30 columns]"
            ]
          },
          "execution_count": 165,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "gencc_db.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 166,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bEk8WeB_7q8w",
        "outputId": "0843635c-cea3-4a9a-c23a-e8cb37558213"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Autosomal recessive                                                   10477\n",
              "Autosomal dominant                                                     8910\n",
              "X-linked                                                               1095\n",
              "Unknown                                                                 892\n",
              "X-linked recessive                                                      144\n",
              "Semidominant                                                            133\n",
              "Mitochondrial                                                           100\n",
              "X-linked dominant                                                        26\n",
              "Somatic mosaicism                                                         9\n",
              "Autosomal dominant inheritance with paternal imprinting                   5\n",
              "Autosomal dominant inheritance with maternal imprinting HP:0012275        4\n",
              "Y-linked inheritance                                                      2\n",
              "Digenic inheritance HP:0010984                                            1\n",
              "Name: moi_title, dtype: int64"
            ]
          },
          "execution_count": 166,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "gencc_db['moi_title'].value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 167,
      "metadata": {
        "id": "zgLByx6jBeJj"
      },
      "outputs": [],
      "source": [
        "mapping = {\n",
        "    'Autosomal dominant': 'AD',\n",
        "    'Autosomal recessive': 'AR',\n",
        "}\n",
        "\n",
        "\n",
        "# Replace values in the 'moi_title' column\n",
        "gencc_db['moi_title'] = gencc_db['moi_title'].map(lambda x: mapping.get(x, 'not found'))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 169,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "o-GwB_6DBJF6",
        "outputId": "a21c916b-808f-4403-fc4b-68ac9c731ad7"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "1067it [00:05, 203.79it/s]\n"
          ]
        }
      ],
      "source": [
        "add_new_data(df, gencc_db, 'gene_symbol', 'moi_title')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 170,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XN2dYHayCDp8",
        "outputId": "85f59b9d-0547-4ace-a30d-f668826e83e7"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Unknown    716\n",
              "AR         175\n",
              "AD         141\n",
              "AD/AR       35\n",
              "Name: MOI, dtype: int64"
            ]
          },
          "execution_count": 170,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['MOI'].value_counts()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RzDP57JxRFvO"
      },
      "source": [
        "# **Loading P_ADs and adding them to df**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 172,
      "metadata": {
        "id": "iHql793_RFej"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "\n",
        "p_ad_data_list = []\n",
        "\n",
        "# Initialize an empty DataFrame to store the data\n",
        "p_ad_df = pd.DataFrame(columns=['HGNC', 'p_ad'])\n",
        "\n",
        "# Loop through each text file\n",
        "for i in range(100, 1001, 100):\n",
        "    filename = f\"{i}.txt\"\n",
        "    filepath = os.path.join(\"/content/drive/MyDrive/pardisgene/P_AD/\", filename)  # Update the path to your files\n",
        "    with open(filepath, 'r') as file:\n",
        "        lines = file.readlines()\n",
        "        # Iterate through the lines in the file\n",
        "        for j in range(0, len(lines), 6):\n",
        "            # Extract the first and fifth lines of each section\n",
        "            gene_name = lines[j].strip()\n",
        "            score = float(lines[j + 4].strip())\n",
        "            # Append the data to the DataFrame\n",
        "            p_ad_data_list.append({'HGNC': gene_name, 'p_ad': score})\n",
        "\n",
        "# Print the DataFrame\n",
        "p_ad_df = pd.concat([p_ad_df, pd.DataFrame(p_ad_data_list)], ignore_index=True)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 173,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "oGg7D6CQTFfn",
        "outputId": "5583958f-81eb-4fa0-898f-e41a50649dd7"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"p_ad_df\",\n  \"rows\": 947,\n  \"fields\": [\n    {\n      \"column\": \"HGNC\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"PGAP3\",\n          \"YPEL2\",\n          \"CD300LB\"\n        ],\n        \"num_unique_values\": 947,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"p_ad\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.3438173364823938,\n        \"min\": 0.0,\n        \"max\": 1.0,\n        \"samples\": [\n          0.876,\n          0.686,\n          0.716\n        ],\n        \"num_unique_values\": 452,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "p_ad_df"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-bbf28ccf-962d-455f-876c-3cf71021ec76\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>HGNC</th>\n",
              "      <th>p_ad</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>AANAT</td>\n",
              "      <td>0.068</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>AARSD1</td>\n",
              "      <td>0.099</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>AATF</td>\n",
              "      <td>0.097</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>AATK</td>\n",
              "      <td>0.205</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ABCA10</td>\n",
              "      <td>0.125</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-bbf28ccf-962d-455f-876c-3cf71021ec76')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-bbf28ccf-962d-455f-876c-3cf71021ec76 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-bbf28ccf-962d-455f-876c-3cf71021ec76');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-c8edf4f5-2d54-4cb8-b2e5-46ae181a17e2\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c8edf4f5-2d54-4cb8-b2e5-46ae181a17e2')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-c8edf4f5-2d54-4cb8-b2e5-46ae181a17e2 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "     HGNC   p_ad\n",
              "0   AANAT  0.068\n",
              "1  AARSD1  0.099\n",
              "2    AATF  0.097\n",
              "3    AATK  0.205\n",
              "4  ABCA10  0.125"
            ]
          },
          "execution_count": 173,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "p_ad_df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 174,
      "metadata": {
        "id": "MfkNFbbBTeEJ"
      },
      "outputs": [],
      "source": [
        "# Merge the two DataFrames based on 'HGNC' and 'Approved Gene Symbol' columns\n",
        "merged_df = pd.merge(df, p_ad_df, left_on='Approved Gene Symbol', right_on='HGNC', how='left')\n",
        "\n",
        "# Drop the 'HGNC' column from the merged DataFrame\n",
        "merged_df.drop(columns=['HGNC'], inplace=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 175,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 311
        },
        "id": "FuDjsMNNUv8l",
        "outputId": "7548513a-f8d3-471a-be7b-60b2e3802514"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"merged_df\",\n  \"rows\": 1067,\n  \"fields\": [\n    {\n      \"column\": \"Chromosome\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"chr17\"\n        ],\n        \"num_unique_values\": 1,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genomic Position Start\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 25050618,\n        \"min\": 1,\n        \"max\": 83079609,\n        \"samples\": [\n          41866917\n        ],\n        \"num_unique_values\": 1019,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genomic Position End\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 25141352,\n        \"min\": 181650,\n        \"max\": 83257441,\n        \"samples\": [\n          59215230\n        ],\n        \"num_unique_values\": 1016,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Cyto Location\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"17q24.3\"\n        ],\n        \"num_unique_values\": 101,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Computed Cyto Location\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"17q25.1\"\n        ],\n        \"num_unique_values\": 38,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MIM Number\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"610465\"\n        ],\n        \"num_unique_values\": 1067,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene/Locus And Other Related Symbols\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"ACSF2, FLJ20920\"\n        ],\n        \"num_unique_values\": 1067,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene Name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"Micro RNA 22\"\n        ],\n        \"num_unique_values\": 1066,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Approved Gene Symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"DNAH17\"\n        ],\n        \"num_unique_values\": 1006,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Entrez Gene ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 17776538.963555794,\n        \"min\": 15.0,\n        \"max\": 117204002.0,\n        \"samples\": [\n          1366.0\n        ],\n        \"num_unique_values\": 1041,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Ensembl Gene ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"ENSG00000006744\"\n        ],\n        \"num_unique_values\": 989,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Comments\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"mutation identified in 1 MGORS5 patient\"\n        ],\n        \"num_unique_values\": 151,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Phenotypes\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"Osteogenesis imperfecta, type II, 166210 (3), Autosomal dominant; Caffey disease, 114000 (3), Autosomal dominant; Ehlers-Danlos syndrome, arthrochalasia type, 1, 130060 (3), Autosomal dominant; Osteogenesis imperfecta, type I, 166200 (3), Autosomal dominant; {Bone mineral density variation QTL, osteoporosis}, 166710 (3), Autosomal dominant; Combined osteogenesis imperfecta and Ehlers-Danlos syndrome 1, 619115 (3), Autosomal dominant; Osteogenesis imperfecta, type IV, 166220 (3), Autosomal dominant; Osteogenesis imperfecta, type III, 259420 (3), Autosomal dominant\"\n        ],\n        \"num_unique_values\": 352,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Mouse Gene Symbol/ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"samples\": [\n          \"Ube2o (MGI:2444266)\"\n        ],\n        \"num_unique_values\": 932,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MOI\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"samples\": [\n          \"AD\"\n        ],\n        \"num_unique_values\": 4,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"p_ad\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.34373090317295796,\n        \"min\": 0.0,\n        \"max\": 1.0,\n        \"samples\": [\n          0.139\n        ],\n        \"num_unique_values\": 450,\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "merged_df"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-f384549c-f654-467a-aa81-830635ffeb67\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Genomic Position Start</th>\n",
              "      <th>Genomic Position End</th>\n",
              "      <th>Cyto Location</th>\n",
              "      <th>Computed Cyto Location</th>\n",
              "      <th>MIM Number</th>\n",
              "      <th>Gene/Locus And Other Related Symbols</th>\n",
              "      <th>Gene Name</th>\n",
              "      <th>Approved Gene Symbol</th>\n",
              "      <th>Entrez Gene ID</th>\n",
              "      <th>Ensembl Gene ID</th>\n",
              "      <th>Comments</th>\n",
              "      <th>Phenotypes</th>\n",
              "      <th>Mouse Gene Symbol/ID</th>\n",
              "      <th>MOI</th>\n",
              "      <th>p_ad</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>10800000</td>\n",
              "      <td>17p13</td>\n",
              "      <td>NaN</td>\n",
              "      <td>608631</td>\n",
              "      <td>ASPG2</td>\n",
              "      <td>Asperger syndrome, susceptibility to, 2</td>\n",
              "      <td>NaN</td>\n",
              "      <td>431711.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>breakpoints between CHRNE and GP1BA</td>\n",
              "      <td>{Asperger syndrome susceptibility 2}, 608631 (...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>113721</td>\n",
              "      <td>BCPR</td>\n",
              "      <td>Breast cancer-related regulator of TP53</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Breast cancer (1)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>10800000</td>\n",
              "      <td>17p13</td>\n",
              "      <td>NaN</td>\n",
              "      <td>601202</td>\n",
              "      <td>CTRCT24, CTAA2</td>\n",
              "      <td>Cataract 24, anterior polar</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1484.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Cataract 24, anterior polar, 601202 (2), Autos...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AD</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>615674</td>\n",
              "      <td>DDD3</td>\n",
              "      <td>Dowling-Degos disease 3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>102997065.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>max lod at D17S1529</td>\n",
              "      <td>Dowling-Degos disease 3, 615674 (2), Autosomal...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AD</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr17</td>\n",
              "      <td>1</td>\n",
              "      <td>3400000</td>\n",
              "      <td>17p13.3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>613215</td>\n",
              "      <td>DUP17p13.3, C17DUPp13.3</td>\n",
              "      <td>Chromosome 17p13.3 duplication syndrome</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>includes LIS1 and/or YWHAE</td>\n",
              "      <td>Chromosome 17p13.3 duplication syndrome, 61321...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Unknown</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f384549c-f654-467a-aa81-830635ffeb67')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-f384549c-f654-467a-aa81-830635ffeb67 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-f384549c-f654-467a-aa81-830635ffeb67');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-f822129d-3210-463b-ab22-c82ec5a0f293\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-f822129d-3210-463b-ab22-c82ec5a0f293')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-f822129d-3210-463b-ab22-c82ec5a0f293 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "  Chromosome  Genomic Position Start  Genomic Position End Cyto Location  \\\n",
              "0      chr17                       1              10800000         17p13   \n",
              "1      chr17                       1               3400000       17p13.3   \n",
              "2      chr17                       1              10800000         17p13   \n",
              "3      chr17                       1               3400000       17p13.3   \n",
              "4      chr17                       1               3400000       17p13.3   \n",
              "\n",
              "  Computed Cyto Location MIM Number Gene/Locus And Other Related Symbols  \\\n",
              "0                    NaN     608631                                ASPG2   \n",
              "1                    NaN     113721                                 BCPR   \n",
              "2                    NaN     601202                       CTRCT24, CTAA2   \n",
              "3                    NaN     615674                                 DDD3   \n",
              "4                    NaN     613215              DUP17p13.3, C17DUPp13.3   \n",
              "\n",
              "                                 Gene Name Approved Gene Symbol  \\\n",
              "0  Asperger syndrome, susceptibility to, 2                  NaN   \n",
              "1  Breast cancer-related regulator of TP53                  NaN   \n",
              "2              Cataract 24, anterior polar                  NaN   \n",
              "3                  Dowling-Degos disease 3                  NaN   \n",
              "4  Chromosome 17p13.3 duplication syndrome                  NaN   \n",
              "\n",
              "   Entrez Gene ID Ensembl Gene ID                             Comments  \\\n",
              "0        431711.0             NaN  breakpoints between CHRNE and GP1BA   \n",
              "1             NaN             NaN                                  NaN   \n",
              "2          1484.0             NaN                                  NaN   \n",
              "3     102997065.0             NaN                  max lod at D17S1529   \n",
              "4             NaN             NaN           includes LIS1 and/or YWHAE   \n",
              "\n",
              "                                          Phenotypes Mouse Gene Symbol/ID  \\\n",
              "0  {Asperger syndrome susceptibility 2}, 608631 (...                  NaN   \n",
              "1                                  Breast cancer (1)                  NaN   \n",
              "2  Cataract 24, anterior polar, 601202 (2), Autos...                  NaN   \n",
              "3  Dowling-Degos disease 3, 615674 (2), Autosomal...                  NaN   \n",
              "4  Chromosome 17p13.3 duplication syndrome, 61321...                  NaN   \n",
              "\n",
              "       MOI  p_ad  \n",
              "0  Unknown   NaN  \n",
              "1  Unknown   NaN  \n",
              "2       AD   NaN  \n",
              "3       AD   NaN  \n",
              "4  Unknown   NaN  "
            ]
          },
          "execution_count": 175,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "merged_df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 176,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "y7zUinLTYr_r",
        "outputId": "bc8e7ac2-5c06-4796-ff1c-287a80a04ba9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "number of NAN in p_ad column: 126\n",
            "number of values in p_ad column: 941\n"
          ]
        }
      ],
      "source": [
        "print(f'number of NAN in p_ad column: {merged_df[\"p_ad\"].isna().sum()}')\n",
        "print(f'number of values in p_ad column: {merged_df[\"p_ad\"].notna().sum()}')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 177,
      "metadata": {
        "id": "lmOkMJHF94d-"
      },
      "outputs": [],
      "source": [
        "merged_df.to_csv('/content/drive/MyDrive/pardisgene/datasets/final_df.csv', index=False)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tV6yLdF8-QFA"
      },
      "source": [
        "# **Starting second part**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "e3DnhmBEq-Dd"
      },
      "outputs": [],
      "source": [
        "def generate_final_output(df : pd.DataFrame, vcf_df: pd.DataFrame) -> pd.DataFrame:\n",
        "  \"\"\"\n",
        "    Generates a final output DataFrame based on matching criteria between two input DataFrames.\n",
        "\n",
        "    Parameters:\n",
        "        df : pd.DataFrame\n",
        "            DataFrame containing genomic position information and associated data.\n",
        "        vcf_df: pd.DataFrame\n",
        "            DataFrame representing variant call data.\n",
        "\n",
        "    Returns:\n",
        "        pd.DataFrame\n",
        "            Final output DataFrame containing merged information.\n",
        "\n",
        "    This function takes two DataFrames as input: 'df' containing genomic position information\n",
        "    and associated metadata, and 'vcf_df' containing variant call data. It iterates through each\n",
        "    row of 'vcf_df' and identifies matching rows in 'df' based on genomic position. For each match,\n",
        "    it constructs a new data dictionary containing relevant information from both DataFrames,\n",
        "    including the variant ID, associated gene symbol, mode of inheritance, and probability of\n",
        "    association (P_AD). If no match is found for a row in 'vcf_df', the corresponding entry in\n",
        "    the output DataFrame contains null values. The function then concatenates all new data\n",
        "    dictionaries into a final output DataFrame and returns it.\n",
        "    \"\"\"\n",
        "  vcf_df_output = pd.DataFrame(columns=['ID', 'SYMBOL', 'MOI', 'P_AD'])\n",
        "\n",
        "  new_data_list = []\n",
        "\n",
        "  # Iterate over each row in vcf_df\n",
        "  for _, row in vcf_df.iterrows():\n",
        "      # Get the POS value from vcf_df\n",
        "      pos = row['POS']\n",
        "\n",
        "      # Check if pos falls within the range defined by 'Genomic Position Start' and 'Genomic Position End' in df\n",
        "      matching_rows = df[(df['Genomic Position Start'] <= pos) & (df['Genomic Position End'] >= pos)]\n",
        "\n",
        "      # Check if there are multiple matching rows in df\n",
        "      if not matching_rows.empty:\n",
        "          for _, matching_row in matching_rows.iterrows():\n",
        "              new_data = {'ID': row['ID'],\n",
        "                          'SYMBOL': matching_row['Gene/Locus And Other Related Symbols'],\n",
        "                          # 'SYMBOL': matching_row['Approved Gene Symbol'],\n",
        "                          'MOI': matching_row['MOI'],\n",
        "                          'P_AD': matching_row['p_ad']}\n",
        "\n",
        "              new_data_list.append(new_data)\n",
        "      else:\n",
        "          new_data = {'ID': row['ID'],\n",
        "                      'SYMBOL': None,\n",
        "                      'MOI': None,\n",
        "                      'P_AD': None}\n",
        "\n",
        "          new_data_list.append(new_data)\n",
        "\n",
        "\n",
        "  vcf_df_output = pd.concat([vcf_df_output, pd.DataFrame(new_data_list)], ignore_index=True)\n",
        "\n",
        "  return vcf_df_output"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 132,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kL2narSNsJRD",
        "outputId": "0d7cdbfc-15f2-4866-8888-58e12b574862"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1179.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1179.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1362.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1362.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1441.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1441.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRA1567.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRA1567.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRA1379.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRA1379.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1559.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1559.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1501.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1501.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1510.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1510.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRA1596.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRA1596.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1989.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1989.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1846.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1846.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRC1100.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRC1100.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1271.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1271.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPVRA1017.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPVRA1017.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA2052.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA2052.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERA1458.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERA1458.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRC1332.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRC1332.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1109.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1109.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPERC1822.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPERC1822.csv\n",
            "Processed file /content/drive/MyDrive/pardisgene/vcf/SMPQRA1257.hg38.vcf saved as /content/drive/MyDrive/pardisgene/outputs/non_match/SMPQRA1257.csv\n"
          ]
        }
      ],
      "source": [
        "from pathlib import Path\n",
        "\n",
        "df = pd.read_csv('/content/drive/MyDrive/pardisgene/datasets/final_df.csv')\n",
        "\n",
        "# Define the directory containing the VCF files\n",
        "vcf_directory = Path('/content/drive/MyDrive/pardisgene/vcf')  # Update with the path to your directory\n",
        "\n",
        "# Define the directory to save the outputs\n",
        "output_directory = Path('/content/drive/MyDrive/pardisgene/outputs')  # Update with the path to your output directory\n",
        "\n",
        "# Iterate over each VCF file in the directory\n",
        "for vcf_file in vcf_directory.glob('SMP*.hg38.vcf'):\n",
        "    # Read the VCF file\n",
        "    if str(vcf_file) in ['/content/drive/MyDrive/pardisgene/vcf/SMPERA1989.hg38.vcf',\n",
        "                    '/content/drive/MyDrive/pardisgene/vcf/SMPERA2052.hg38.vcf']:\n",
        "\n",
        "      vcf_df = pd.read_csv(vcf_file , sep='\\t', skiprows=23)\n",
        "\n",
        "    else:\n",
        "      vcf_df = pd.read_csv(vcf_file , sep='\\t', skiprows=24)\n",
        "\n",
        "\n",
        "    try:\n",
        "      final_output_df = generate_final_output(df, vcf_df)\n",
        "\n",
        "      # Save the processed data to the output directory\n",
        "      output_file_path = f'{output_directory}/{vcf_file.stem.split(\".\")[0]}.csv'\n",
        "      final_output_df.to_csv(output_file_path, index=False)\n",
        "\n",
        "\n",
        "      print(f\"Processed file {vcf_file} saved as {output_file_path}\")\n",
        "    except Exception as e:\n",
        "      print(f\"Error processing file {vcf_file}: {e}\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 122,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uYtBfPKWJbUQ",
        "outputId": "7baacc75-b6b2-4cac-adc6-5362d8e50863"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "MOI\n",
              "Unknown    8293\n",
              "AD         3485\n",
              "AR         1563\n",
              "AD/AR       110\n",
              "dtype: int64"
            ]
          },
          "execution_count": 122,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "final_output_df.value_counts('MOI')"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
